src/ynEIGEN/ynEigenViewer.sol

```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: BSD 3-Clause License
+pragma solidity ^0.8.24;
+
+import {IERC20, IERC20Metadata} from "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+
+import {IynEigen} from "../interfaces/IynEigen.sol";
+import {IRateProvider} from "../interfaces/IRateProvider.sol";
+import {ITokenStakingNodesManager,ITokenStakingNode} from "../interfaces/ITokenStakingNodesManager.sol";
+
+import {AssetRegistry} from "./AssetRegistry.sol";
+
+contract ynEigenViewer {
+    
+    struct AssetInfo {
+        address asset;
+        string name;
+        string symbol;
+        uint256 ratioOfTotalAssets;
+        uint256 totalBalance;
+    }
+    
+    AssetRegistry public immutable assetRegistry;
+    IynEigen public immutable ynEIGEN;
+    ITokenStakingNodesManager public immutable tokenStakingNodesManager;
+    IRateProvider public immutable rateProvider;
+
+    uint256 public constant DECIMALS = 1_000;
+    uint256 public constant UNIT = 1 ether;
+
+    constructor(address _assetRegistry, address _ynEIGEN, address _tokenStakingNodesManager, address _rateProvider) {
+        assetRegistry = AssetRegistry(_assetRegistry);
+        ynEIGEN = IynEigen(_ynEIGEN);
+        tokenStakingNodesManager = ITokenStakingNodesManager(_tokenStakingNodesManager);
+        rateProvider = IRateProvider(_rateProvider);
+    }
+
+    function getAllStakingNodes() external view returns (ITokenStakingNode[] memory) {
+        return tokenStakingNodesManager.getAllNodes();
+    }
+
+    function getYnLSDeAssets() external view returns (AssetInfo[] memory _assetsInfo) {
```

Comment:
lets rename this to ynEigen to be able to reuse the contract without refactoring for other ynXXXe tokens

Reply:
https://github.com/yieldnest/yieldnest-protocol/pull/145/commits/9b9e9dad60fe56d9ade43f2316d429bf7e801f53